File: cpp/54a_1833A_MusicalPuzzle.cpp
#include<bits/stdc++.h>

int solve(std::string s, int n){
    std::unordered_set<std::string> hset;
    hset.insert(s.substr(0, 2));
    for(int i = 2; i < n; i++){
        hset.insert(s.substr(i-1, 2));
    }
    return hset.size();
}

int main(){
    int t, n;
    std::string s;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        std::cin>>s;
        ans.push_back(solve(s, n));
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/54a_1833A_MusicalPuzzle.py
def solve(s, n):
    hset = set()
    for i in range(1, n):
        hset.add(s[i-1:i+1])
    return len(hset)

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input()
        ans.append(solve(s, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()